iavl/iavl_proof.go

```diff
@@ -101,6 +101,14 @@ func (leaf IAVLProofLeafNode) Hash() []byte {
 	return hasher.Sum(nil)
 }
 
+func (leaf IAVLProofLeafNode) isLesserThan(key []byte) bool {
+	return bytes.Compare(leaf.KeyBytes, key) == -1
+}
+
+func (leaf IAVLProofLeafNode) isGreaterThan(key []byte) bool {
+	return bytes.Compare(key, leaf.KeyBytes) == -1
```

Comment:
I think keeping leaf.KeyBytes as the subject for bytes.Compare helps with readability so I'd suggest
```go
return bytes.Compare(leaf.KeyBytes, key) == 1
```
and then somewhere some unit tests for these two functions, please.